legend-engine-xt-sql-postgres-server/src/main/java/org/finos/legend/engine/postgres/Session.java

```diff
@@ -103,16 +105,31 @@ public void parse(String statementName, String query, List<Integer> paramTypes)
     private SessionHandler getSessionHandler(String query)
     {
         SqlBaseParser parser = SQLGrammarParser.getSqlBaseParser(query, "query");
-        List<QualifiedName> qualifiedNames = EXTRACTOR.visitSingleStatement(parser.singleStatement());
+        SqlBaseParser.SingleStatementContext ctx = parser.singleStatement();
+        SqlBaseParser.StatementContext statement = ctx.statement();
 
-        boolean isMetadataQuery = qualifiedNames.stream().flatMap(i -> i.parts.stream()).anyMatch(SystemSchemas::contains);
-        if (isMetadataQuery)
+        if (statement instanceof SqlBaseParser.DefaultContext)
         {
-            return metaDataSessionHandler;
+            SqlBaseParser.DefaultContext defaultContext = (SqlBaseParser.DefaultContext) statement;
+            List<QualifiedName> qualifiedNames = EXTRACTOR.visitDefault(defaultContext);
+            boolean isMetadataQuery = qualifiedNames.isEmpty() || qualifiedNames.stream().flatMap(i -> i.parts.stream()).anyMatch(SystemSchemas::contains);
+            if (isMetadataQuery)
+            {
+                return metaDataSessionHandler;
+            }
+            else
+            {
+                return dataSessionHandler;
+            }
+        }
+        else if (statement instanceof SqlBaseParser.SetContext)
+        {
+            // TODO: Handle SET queries
+            return emptySessionHandler;
         }
         else
         {
-            return dataSessionHandler;
+            return metaDataSessionHandler;
```

Comment:
@gs-manvig As discussed, let's create a follow on issue to make this more defensive to report statements that we don't support / recognize (rather than just returning default empty success)